vortex-array/src/array/varbin/builder.rs

```diff
@@ -80,7 +82,11 @@ impl<O: NativePType> VarBinBuilder<O> {
 
     pub fn finish(mut self, dtype: DType) -> VarBinArray {
         let offsets = PrimitiveArray::from(self.offsets);
-        let data = PrimitiveArray::from_bytes(self.data.freeze(), Validity::NonNullable);
+        let data = PrimitiveArray::new(
+            Buffer::from(Bytes::from_owner(self.data.into_shared())),
```

Comment:
This should disappear when we port vortex-buffer over to using internally aligned buffers